src/azul/service/manifest_service.py

```diff
@@ -750,8 +750,16 @@ def _get_cached_manifest_file_name(self,
                 try:
                     encoded_file_name = tagging[self.file_name_tag]
                 except KeyError:
-                    # Can't be absent under S3's strong consistency
-                    assert False, (object_key, self.file_name_tag)
+                    # Paged manifest generators apply the tag after creating
+                    # the object. If we happen to be right in between the
+                    # creation and application of tags, we pretend that the
+                    # manifest object doesn't exist, assuming that the caller
+                    # will tell the client to check back later. Note that this
+                    # involves the attempted creation of step-function
+                    # execution, but because step-functions generate manifests
+                    # idempotentently, subsequent requests for the same
+                    # manifest are associated to a single SF execution.
```

Comment:
```suggestion
                    # While unpaged manifest generators apply the tag *at* object creation, paged ones do so in a separate request. If we happen to get here right in between the
                    # creation and application of tags, we pretend that the
                    # manifest object doesn't exist, relying on client to check back later. Note that this
                    # involves the attempted creation of a step-function (SF)
                    # execution, but because SFs generate manifests
                    # idempotently, subsequent requests for the same
                    # manifest are associated with a single SF execution.
```

Reply:
I don't understand the second sentence. Please raise in PL.